src/utils/slpParser.ts

```diff
@@ -266,3 +282,42 @@ export class SlpParser extends EventEmitter {
     }
   }
 }
+
+class RollbackCounter {
```

Comment:
Can we move this into a separate class? Perhaps `src/utils/RollbackCounter.ts` or something?